pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -114,6 +125,15 @@ func validateRuleNode(
 	return &newAlertRule, nil
 }
 
+func validateLabels(l map[string]string) error {
```

Comment:
Yes, this would be a good place to eventually centralize this logic but it would be a functional change for some users so should be kept separate to this PR IMO.